Source/Libraries/FaultData/DataResources/FaultDataResource.cs

```diff
@@ -1023,14 +1023,80 @@ private void PopulateFaultInfo(Fault fault, DataGroup dataGroup, VICycleDataGrou
 
             if (calculationCycle >= 0)
             {
+                TableOperations<Meter> meterTable = new TableOperations<Meter>(connection);
+                string testQuery = "ID={0} AND LocationID in (Select LocationID from AssetLocation where AssetID={1})";
                 for (int i = 0; i < fault.Curves.Count; i++)
                 {
                     Fault.Summary summary = new Fault.Summary();
                     summary.DistanceAlgorithmIndex = i;
                     summary.DistanceAlgorithm = fault.Curves[i].Algorithm;
                     summary.Distance = fault.Curves[i][calculationCycle].Value;
-                    summary.IsValid = IsValid(summary.Distance, dataGroup);
+                    Line line = Line.DetailedLine(dataGroup.Asset);
+                    List<Fault.Path> faultPaths = new List<Fault.Path>();
+                    for (int index = 0; index < line.Path.Count; index++)
+                    {
+                        // We can break early if this condition is found because any segments after are shorter
+                        if (summary.Distance > line.Path[index].Length) break;
```

Comment:
This doesn't seem to be true. Paths appear to be arbitrarily ordered on account of the list having been generated by enumerating a `Dictionary` without any explicit sorting.

https://github.com/GridProtectionAlliance/openXDA/blob/c5377ffa511e6fec38f851823f0ae596f7a42dbf/Source/Libraries/openXDA.Model/TransmissionElements/Line.cs#L194-L203

Reply:
See line 74, it is ordered.